TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,228 @@
+## Overview of CProver Tools
+
+This document provides a brief overview of the various tools in
+the CProver project. The goal is as to provide a lightweight reference
+with links to detailed documentation on each of the tools.
+
+The tools in the CProver project are as follows:
+- [cbmc](#cbmc)
+- [goto-analyzer](#goto-analyzer)
+- [goto-cc (goto-gcc, goto-ld)](#goto-cc)
+- [goto-diff](#goto-diff)
+- [goto-harness](#goto-harness)
+- [goto-instrument](#goto-instrument)
+- [janalyzer](#janalyzer)
+- [java-unit](#java-unit)
+- [jbmc](#jbmc)
+- [jdiff](#jdiff)
+- [memory-analyzer](#memory-analyzer)
+- [smt2_solver](#smt2_solver)
+- [symtab2gb](#symtab2gb)
+- [unit](#unit)
+- [others (converter, driver, file_converter, java-converter)](#others)
+
+The rest of this document provides a section on each of these tools in alphabetical order.
+Most links in this document are to the [CProver online documentation](http://cprover.diffblue.com/index.html).
```

Comment:
There is no consistent way of linking.

I was working on another (draft, so very provisional) pull request (see https://github.com/diffblue/cbmc/pull/5781) and it was recommended to use links to the online documentation at diffblue.com. So should I document with:
A. Links to the local documentation path (e.g. doc/html/index.html)
B. Links to the absolute location (e.g. http://cprover.diffblue.com/index.html)
C. Something else? (E.g. relative html links, although these will not work when .md files are used like this one?)

Reply:
I think the confusion is that the top-level .md files are currently not compiled into cprover.diffblue.com, but only those in `doc` and the `README.md` files in the source directories.

Reply:
My rationale was that this is a top level .md file (similar to README.md) and so we should link to the diffblue.com URLs. Since this is not going to be compiled into the website, then I understand the current linking is correct?

The above said, if we wish to move this document to another location (or have another reason to change), let me know and I can redo the links as relative.